### PR TITLE
MINOR: Update Scala to 2.12.11

### PR DESCRIPTION
**Pull request by ijuma**
Highlights:
* Performance improvements in the ollections
library: algorithmic improvements and
changes to avoid unnecessary allocations.
* Performance improvements in the compiler.
* ASM was upgraded to 7.3.1, allowing the
optimizer to run on JDK 13+.

Full release notes: https://github.com/scala/scala/releases/tag/v2.12.11

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -29,7 +29,7 @@ ext {
 }
 
 // Add Scala version
-def defaultScala212Version = '2.12.10'
+def defaultScala212Version = '2.12.11'
```
Comment (omkreddy) [inline gradle/dependencies.gradle:32]: we should also update `gradle.properties`, `bin/kafka-run-class.sh`, `bin/windows/kafka-run-class.bat`?

**Comment by omkreddy** [comment]
spotbug failure for Scala 2.12  build.

File: gradle/dependencies.gradle (hunk 1 above)
Comment (ijuma) [reply gradle/dependencies.gradle:32]: Yes.

**Comment by ijuma** [comment]
@omkreddy Thanks for the review, updated the PR.

**Review by omkreddy — APPROVED** [review]
@ijuma Thanks for the PR. LGTM.

**Review by chia7712 — APPROVED** [review]

**Comment by ijuma** [comment]
One job passed, one failed with a known flake:

> kafka.api.ConsumerBounceTest.testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup

**Pull request merged**